api/datasets.go

```diff
@@ -34,179 +33,87 @@ func NewDatasetHandlers(inst *lib.Instance, readOnly bool) *DatasetHandlers {
 	return &h
 }
 
-// ListHandler is a dataset list endpoint
-func (h *DatasetHandlers) ListHandler(w http.ResponseWriter, r *http.Request) {
-	switch r.Method {
-	case http.MethodGet, http.MethodPost:
-		if h.ReadOnly {
-			readOnlyResponse(w, "/list")
-			return
-		}
-		h.listHandler(w, r)
-	default:
-		util.NotFoundHandler(w, r)
-	}
-}
+// SaveHandler is a dataset save/update endpoint
+func (h *DatasetHandlers) SaveHandler(routePrefix string) http.HandlerFunc {
+	handleChanges := h.saveHandler(routePrefix)
```

Comment:
`handleChanges` -> `handleSave`, but instead of doing that, I'd prefer we refactor this `SaveHandler` + `saveHandler` into a single method that inlines the `saveHandler` method body into the exported `SaveHandler`.

Instead of a `switch` on the HTTP method, `if r.Method != http.MethodPut && r.Method != http.MethodPost` early-return an error. with bad HTTP verbs handled the rest of the function can be the contents of `saveHandler`. Whenever we have these wrapping funds that just check for one or two HTTP verbs & then call an unexported method, we should combine them

Reply:
omg I thought `handleChanges` was some pattern we were using, because I saw it outside of the `ChangeHandler`... but now that seems really obvious hahah.